lib/SPIRV/libSPIRV/SPIRVError.h

```diff
@@ -174,6 +174,18 @@ inline bool SPIRVErrorLog::checkError(bool Cond, SPIRVErrorCode ErrCode,
   return Cond;
 }
 
+// Return the message associated with the error code or empty string if the
+// error code is invalid
+inline std::string getErrorMessage(int ErrCode) {
+  std::string ErrorMessage;
+  // It's ok if the error code is not present in the map, so we don't need to
```

Comment:
Why is it ok if getErrorMessage is called with an invalid error code?  What if the programmer makes a mistake and passes a wrong int variable to getErrorMessage?  getErrorMessage 
return an empty string which is the same string returned by the Success error code:

> _SPIRV_OP(Success, "")

Reply:
The goal of a new API call is to return a human-readable info about an error code. It's not aimed to be a check whether the argument is an error or success -- the programmer uses the integer value for such a check. This makes ambiguity of the empty string less important. However, if we really wished to address the issue of returning an empty string for both Success error code and for a missing error code, I'd say it's a problem of a choice of the value for _SPIRV_OP(Success, ""). Personally, I'd vote for changing it into _SPIRV_OP(Success, "Success") that both looks logical and resolves the issue of distinguishing between Success error code and a missing error code.

Reply:
I would prefer changing success to "Success" and also changing an invalid error code to something like "Invalid error code input".   "Invalid error code input" will let the user more quickly understand what is happening than returning an empty string.   I think it also makes the code clearer.  When I see the comment:

>  // It's ok if the error code is not present in the map

it is unsettling to me as a code reader.  I will think: "Why should it be OK if the error code is not in the map?  This is not what I would normally expect, so is there some property of the code that I am missing."